From b65b976d23ba12ecf86919bfcf61d807d0118e45 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Thu, 25 Jan 2018 16:40:21 +0800
Subject: [PATCH 33/33] staging: fsl-dpaa2/mac: backport use of 4.9
 ndo_get_stats64

This patch changes the declarations of ndo_get_stats64 handlers
to the previous struct rtnl_link_stats64 * return type instead of
the mainline void return.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 3 ++-
 drivers/staging/fsl-dpaa2/evb/evb.c            | 4 ++--
 drivers/staging/fsl-dpaa2/mac/mac.c            | 5 +++--
 3 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index a8bea3c..edc85f6 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -1396,7 +1396,7 @@ static int dpaa2_eth_set_addr(struct net_device *net_dev, void *addr)
 /** Fill in counters maintained by the GPP driver. These may be different from
  * the hardware counters obtained by ethtool.
  */
-static void dpaa2_eth_get_stats(struct net_device *net_dev,
+static struct rtnl_link_stats64 *dpaa2_eth_get_stats(struct net_device *net_dev,
 				struct rtnl_link_stats64 *stats)
 {
 	struct dpaa2_eth_priv *priv = netdev_priv(net_dev);
@@ -1412,6 +1412,7 @@ static void dpaa2_eth_get_stats(struct net_device *net_dev,
 		for (j = 0; j < num; j++)
 			netstats[j] += cpustats[j];
 	}
+	return stats;
 }
 
 static int dpaa2_eth_change_mtu(struct net_device *net_dev, int mtu)
diff --git a/drivers/staging/fsl-dpaa2/evb/evb.c b/drivers/staging/fsl-dpaa2/evb/evb.c
index 9812b65..dade698 100644
--- a/drivers/staging/fsl-dpaa2/evb/evb.c
+++ b/drivers/staging/fsl-dpaa2/evb/evb.c
@@ -759,7 +759,7 @@ static int evb_dellink(struct net_device *netdev,
 	return 0;
 }
 
-void evb_port_get_stats(struct net_device *netdev,
+static struct rtnl_link_stats64 *evb_port_get_stats(struct net_device *netdev,
 			struct rtnl_link_stats64 *storage)
 {
 	struct evb_port_priv	*port_priv = netdev_priv(netdev);
@@ -836,7 +836,7 @@ void evb_port_get_stats(struct net_device *netdev,
 	if (unlikely(err))
 		goto error;
 
-	return;
+	return storage;
 
 error:
 	netdev_err(netdev, "dpdmux_if_get_counter err %d\n", err);
diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 3ddf344..9b11645 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -164,7 +164,7 @@ static int dpaa2_mac_set_settings(struct net_device *netdev,
 	return phy_ethtool_sset(netdev->phydev, cmd);
 }
 
-static void dpaa2_mac_get_stats(struct net_device *netdev,
+static struct rtnl_link_stats64 *dpaa2_mac_get_stats(struct net_device *netdev,
 				struct rtnl_link_stats64 *storage)
 {
 	struct dpaa2_mac_priv	*priv = netdev_priv(netdev);
@@ -227,9 +227,10 @@ static void dpaa2_mac_get_stats(struct net_device *netdev,
 	if (err)
 		goto error;
 
-	return;
+	return storage;
 error:
 	netdev_err(netdev, "dpmac_get_counter err %d\n", err);
+	return storage;
 }
 
 static struct {
-- 
2.7.4

